### PR TITLE
Test: Check that files in /etc are not zeroed out after docker cp

### PR DESCRIPTION
**Pull request by GabyCT**
Signed-off-by: Cervantes Tellez, Gabriela <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
qa-passed

File: tests/integration/docker/cp.bats [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+@test "Check mounted files at /etc after a docker cp" {
+	skip "currently this test is not working (see https://github.com/01org/cc-oci-runtime/issues/325)"
+	content="test"
+	$DOCKER_EXE run --name containertest -d ubuntu bash -c "sleep 50"
+	echo $content > testfile
+	$DOCKER_EXE cp testfile containertest:/root/
+	$DOCKER_EXE exec -i containertest bash -c "ls /root/ | grep testfile"
+	$DOCKER_EXE exec -i containertest bash -c "egrep "[A-Za-z0-9]+" /etc/resolv.conf"
+	rm -rf testfile
```
Comment (gorozco1) [inline tests/integration/docker/cp.bats]: is a simple file I would suggest to remove "-rf" :S is too aggressive

File: tests/integration/docker/cp.bats [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+SRC="${BATS_TEST_DIRNAME}/../../lib/"
+
+setup() {
+	source $SRC/test-common.bash
+	clean_docker_ps
+	runtime_docker
+}
+
+@test "Check mounted files at /etc after a docker cp" {
```
Comment (gorozco1) [inline tests/integration/docker/cp.bats]: What about big files?

**Comment by GabyCT** [comment]
@gorozco1 changes applied in removing the file :) thanks

File: tests/integration/docker/cp.bats [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	skip "currently this test is not working (see https://github.com/01org/cc-oci-runtime/issues/325)"
+	content="test"
+	$DOCKER_EXE run --name containertest -d ubuntu bash -c "sleep 50"
+	echo $content > testfile
```
Comment (jodh-intel) [inline tests/integration/docker/cp.bats]: What's the working directory when this code is run? I'd be tempted to do something like this to guard against running in an environment where the source directory may not be writable (maybe an out-of-tree build-type environment):

```
testfile=$(mktemp --tmpdir="$BATS_TMPDIR" --suffix=-cor-test)
echo "$content" > "$testfile"
     :
     :
rm -f "$testfile"
```

If you don't like the fact that you are creating, then rewriting the temp file, you could use:

```
test_dir=$(mktemp -d --tmpdir="$BATS_TMPDIR" --suffix=-cor-test)
echo "$content" > "$test_dir/$testfile"
     :
     :
rm -rf "$test_dir"
```

**Comment by GabyCT** [comment]
@jodh-intel Thanks for the feedback, changes were applied

File: tests/integration/docker/cp.bats [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	source $SRC/test-common.bash
+	clean_docker_ps
+	runtime_docker
+}
+
+@test "Check mounted files at /etc after a docker cp" {
+	skip "currently this test is not working (see https://github.com/01org/cc-oci-runtime/issues/325)"
+	content="test"
+	testfile=$(mktemp --tmpdir="$BATS_TMPDIR" --suffix=-cor-test)
+	$DOCKER_EXE run --name containertest -d ubuntu bash -c "sleep 500"
```
Comment (devimc) [inline tests/integration/docker/cp.bats]: ```'-c sleep 500'``` is not needed

**Comment by GabyCT** [comment]
@devimc Thanks Julio for noticing that, I made a typo there

**Comment by GabyCT** [comment]
@devimc I already corrected it

File: tests/integration/docker/cp.bats [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+@test "Check mounted files at /etc after a docker cp" {
+	skip "currently this test is not working (see https://github.com/01org/cc-oci-runtime/issues/325)"
+	content="test"
+	testfile=$(mktemp --tmpdir="$BATS_TMPDIR" --suffix=-cor-test)
+	$DOCKER_EXE run --name containertest -d ubuntu bash -c "sleep 50"
+	echo $content > $testfile
+	$DOCKER_EXE cp $testfile containertest:/root/
+	$DOCKER_EXE exec -i containertest bash -c "ls /root/ | grep cor"
```
Comment (devimc) [inline tests/integration/docker/cp.bats]: you could replace ```"ls /root/ | grep cor"``` with ```"ls /root/$(basename $testfile)"``` to improve the precision of test

File: tests/integration/docker/cp.bats [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	$DOCKER_EXE exec -i containertest bash -c "egrep "[A-Za-z0-9]+" /etc/resolv.conf"
```
Comment (devimc) [inline tests/integration/docker/cp.bats]: you could use ```bash -c "[ -s /etc/resolv.conf ]"```  looks better :)

File: tests/integration/docker/cp.bats [hunk 7]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bats
+# *-*- Mode: sh; sh-basic-offset: 8; indent-tabs-mode: nil -*-*
+
+#  This file is part of cc-oci-runtime.
+#
+#  Copyright (C) 2016 Intel Corporation
```
Comment (grahamwhaley) [inline tests/integration/docker/cp.bats]: (c) 2017 ? :-)

File: tests/integration/docker/cp.bats (hunk 2 above)
Comment (grahamwhaley) [inline tests/integration/docker/cp.bats]: @gorozco1 I'm not sure I understand - are big files different in some significant manner? Can you expand please?

**Comment by jodh-intel** [comment]
Related to #325.

**Comment by GabyCT** [comment]
@devimc and @grahamwhaley changes were applied :)

**Comment by devimc** [comment]
@GabyCT - Thank you!!

**Comment by devimc** [comment]
lgtm

**Comment by grahamwhaley** [comment]
lgtm

**Comment by grahamwhaley** [comment]
@chavafg - I think this PR was updated since the last qa-passed - to be safe I think we may wish to run a qa cycle on this again (I forsee no issues...) what do you think?

**Comment by chavafg** [comment]
yeap, rerunning again :)

**Comment by chavafg** [comment]
qa-passed

File: tests/integration/docker/cp.bats [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+SRC="${BATS_TEST_DIRNAME}/../../lib/"
+
+setup() {
+	source $SRC/test-common.bash
+	clean_docker_ps
+	runtime_docker
+}
+
+@test "Check mounted files at /etc after a docker cp" {
+	skip "currently this test is not working (see https://github.com/01org/cc-oci-runtime/issues/325)"
```
Comment (devimc) [inline tests/integration/docker/cp.bats]: #325  was fixed, please remove this line

**Comment by devimc** [comment]
lgtm

**Comment by devimc** [comment]
rebase it please

**Pull request merged**